server/master/src/main/java/org/apache/accumulo/master/TimeoutTaskExecutor.java

```diff
@@ -121,25 +123,43 @@ public void onTimeout(TimeoutCallback<C> timeoutCallback) {
    * @throws IllegalStateException
    *           If all of the callbacks were not registered before calling this method.
    * @throws InterruptedException
-   *           If interrupted while awaiting callable results.
+   *           If interrupted while awaiting trackingCallable results.
    */
   public void complete() throws InterruptedException {
     Preconditions.checkState(successCallback != null, "Must set a success callback before completing " + this);
     Preconditions.checkState(exceptionCallback != null, "Must set an exception callback before completing " + this);
     Preconditions.checkState(timeoutCallback != null, "Must set a timeout callback before completing " + this);
 
+    while (hasUnfinishedTasks()) {
```

Comment:
It seems like this loop can burn CPU, could sleep a fraction of the timeout.

Reply:
Good call. I opted to use a BlockingQueue so I didn't have to implement my own waiting.